include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -181,11 +181,11 @@ arena_decay_tick(tsdn_t *tsdn, arena_t *arena) {
 
 JEMALLOC_ALWAYS_INLINE void *
 arena_malloc(tsdn_t *tsdn, arena_t *arena, size_t size, szind_t ind, bool zero,
-    tcache_t *tcache, bool slow_path) {
+    bool slab, tcache_t *tcache, bool slow_path) {
 	assert(!tsdn_null(tsdn) || tcache == NULL);
 
 	if (likely(tcache != NULL)) {
-		if (likely(size <= SC_SMALL_MAXCLASS)) {
+		if (likely(size <= SC_SMALL_MAXCLASS && slab)) {
```

Comment:
Similarly here, if we change this to `if (slab) ...`, we may as well just make the other branch an `else if` instead of an `if`, assuming your intention here is clarity. This is instead of adding `assert(!slab)` before `tcache_alloc_large`, which will always be true since you would've returned by calling `tcache_alloc_small` otherwise.

Reply:
Agreed. The "else if" sounds good.